### PR TITLE
set admin.conf ownership to root:root

### PR DESCRIPTION
**Pull request by garcialuis**
#### What type of PR is this?
type::feature (SC-31335)

#### What this PR does / why we need it:
Part of CIS Benchmark Compliance Remediation
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by emosbaugh — COMMENTED** [review]
Does this affect the outro in any way? What I can't figure out is why it was like this before https://github.com/replicatedhq/kURL/blob/dfd58e08e8ad5d300f3d983b7d0aeb13dc794fa2/scripts/common/common.sh#L511-L537

File: scripts/common/common.sh [hunk 1]
```diff
@@ -327,11 +327,9 @@ exportKubeconfig() {
     local kubeconfig
     kubeconfig="$(${K8S_DISTRO}_get_kubeconfig)"
 
-    current_user_sudo_group
-    if [ -n "$FOUND_SUDO_GROUP" ]; then
-        chown root:$FOUND_SUDO_GROUP ${kubeconfig}
-    fi
-    chmod 440 ${kubeconfig}
+    chown root:root ${kubeconfig}
+    chmod 444 ${kubeconfig}
```
Comment (laverya) [inline scripts/common/common.sh:331]: I just know someone's going to complain to us about this, and I think a much better long term solution would be to make a copy of the kubeconfig file that's owned by the sudo user group

but this'll work for the moment

**Review by laverya — DISMISSED** [review]

**Comment by laverya** [comment]
> Does this affect the outro in any way? What I can't figure out is why it was like this before
> 
> https://github.com/replicatedhq/kURL/blob/dfd58e08e8ad5d300f3d983b7d0aeb13dc794fa2/scripts/common/common.sh#L511-L537

Thaaaat is actually exactly the logic that I wanted to see
@garcialuis can we actually set the permisions for admin.conf to `440` and always create a kubeconfig for the user instead?

**Comment by garcialuis** [comment]
> Thaaaat is actually exactly the logic that I wanted to see @garcialuis can we actually set the permisions for admin.conf to `440` and always create a kubeconfig for the user instead?

Okay, yeah - I'll change that back and create a separate kubeconfig.

**Comment by emosbaugh** [comment]
> > Thaaaat is actually exactly the logic that I wanted to see @garcialuis can we actually set the permisions for admin.conf to `440` and always create a kubeconfig for the user instead?
> 
> Okay, yeah - I'll change that back and create a separate kubeconfig.

We will have to maintain that additional copy of the config and update it when the admin.conf changes. I know that certs eventually get rotated before they expire and the admin.conf may change.

**Comment by emosbaugh** [comment]
> Thaaaat is actually exactly the logic that I wanted to see @garcialuis can we actually set the permisions for admin.conf to `440` and always create a kubeconfig for the user instead?

Actually @laverya wont that affect other users other that may log into the machine other than the one that actually installs kurl?

**Comment by laverya** [comment]
> Actually @laverya wont that affect other users other that may log into the machine other than the one that actually installs kurl?

It will, but I think it's still better than letting _every_ user of the instance run kubectl commands as admin

**Review by Wampir0990 — DISMISSED** [review]

File: scripts/common/common.sh (hunk 1 above)
Comment (emosbaugh) [reply scripts/common/common.sh:331]: I think we should put this behind the feature flag that Lavery added in the spec.

Comment (garcialuis) [reply scripts/common/common.sh]: I updated this file now to check for the CIS_COMPLIANCE flag that Andrew introduced

**Review by emosbaugh — COMMENTED** [review]
This looks like it's bringing in a lot of changes.

**Comment by garcialuis** [comment]
Closing this outdate pr, replaced by [2756](https://github.com/replicatedhq/kURL/pull/2756).

**Pull request closed without merging**